modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/query/builder/FilterQueries.scala

```diff
@@ -25,10 +24,17 @@ import ch.epfl.bluebrain.nexus.kg.indexing.{ConfiguredQualifier, Qualifier}
   * @tparam F  the monadic effect type
   * @tparam Id the generic type which defines the response's payload
   */
-class FilterQueries[F[_], Id](queryClient: SparqlQuery[F], querySettings: QuerySettings) {
-
+class FilterQueries[F[_], Id](queryClient: SparqlQuery[F], querySettings: QuerySettings)(
```

Comment:
use the most specific filter only if it includes the less specific ones.

Reply:
Done